app/helpers/payout_helper.rb

```diff
@@ -0,0 +1,73 @@
+module PayoutHelper
+  def icon_class(statuses, current, index)
+    selected = index <= statuses.find_index(current)
+
+    return 'inactive' unless selected
+    return "animated" if  statuses[index] == current && current != I18n.t(".publishers.payout_status.statuses.done")
+
+    "active"
+  end
+
+  def current_status_and_percent(report_created_at)
+    today = DateTime.now
```

Comment:
I prefer refactoring to make it more readable

e.g.
```
status = nil
status = DONE if !Rails.cache.fetch('payout_in_progress')
if status.blank
days_ago = (Date.today - report_created_at.to_date).to_i
if days_ago < 3
  status = PREPARING
  progress_percentage = ((days_ago / 3) * 100).to_i
elsif days_ago < 7
  status = REVIEWING
  progress_percentage = (((days_ago - 3) / 4) * 100).to_i
elsif days_ago < 11
  status = IN_PROGRESS
  progress_percentage = (((days_ago - 7) / 4) * 100).to_i
else
  status = DONE
end

return [I18n.t(".publishers.payout_status.statuses.done"), nil] if status == DONE
```

Reply:
I like this, my original approach was very similar, but I thought having a bunch of if statements might make it confusing - however looks better than the current implementation. 

One of the things I'd eventually want is to be able to display the payout progress tab for a few days after the payout is complete. If the payout in progress was removed on a certain day it would be helpful to automatically switch over to "Done" and show the end users.

Reply:
Probably something in Redis is easiest or adding status to `payout_reports`